### ISSUE TITLE
Bug: Local storage overflow fails silently, without letting the user know

### ISSUE THREAD

**Issue by bgevko** [excalidraw/excalidraw#8805]
Nobody is probably using the app this way, but I was playing around and noticed that if I paste in too many items, local storage gets overflowed. It's not obvious because the error is in the console.

![Image](https://github.com/user-attachments/assets/c5c66200-6594-4cbd-830d-5148e352ca27)

**Comment by moroii69** [comment]
i encountered the same issue where pasting a large number of items triggers a `quotaexceedederror` due to localstorage reaching its size limit. this happens because localstorage has a fixed storage capacity, and large data exceeds that limit.

well, the [excalidraw+ plan](https://plus.excalidraw.com/pricing), which includes **cloud storage**, might solve this issue by storing data externally, thus avoiding the localstorage limitation.

**Comment by akash1520** [comment]
> i encountered the same issue where pasting a large number of items triggers a `quotaexceedederror` due to localstorage reaching its size limit. this happens because localstorage within the browser has a **fixed** storage capacity, and large data exceeds that limit.
> 
> well, the [excalidraw+ plan](https://plus.excalidraw.com/pricing), which includes **cloud storage**, might solve this issue by storing data externally, thus avoiding the localstorage limitation.

If it's going to be tackled by the [excalidraw+ plan](https://plus.excalidraw.com/pricing), then I think we can just give user an alert that your `localstorage` is about to get full and you might lose your work because of that so if you want to avoid that you can register for [excalidraw+ plan](https://plus.excalidraw.com/pricing)

**Comment by moroii69** [comment]
i agree. it'd be helpful to show a notification or toast when storage usage reaches around 80-90% of the total limit. 

so users can be informed before hitting the maximum capacity.

**Comment by moroii69** [comment]
> > i encountered the same issue where pasting a large number of items triggers a `quotaexceedederror` due to localstorage reaching its size limit. this happens because localstorage within the browser has a **fixed** storage capacity, and large data exceeds that limit.
> > 
> > well, the [excalidraw+ plan](https://plus.excalidraw.com/pricing), which includes **cloud storage**, might solve this issue by storing data externally, thus avoiding the localstorage limitation.
> 
> If it's going to be tackled by the [excalidraw+ plan](https://plus.excalidraw.com/pricing), then I think we can just give user an alert that your `localstorage` is about to get full and you might lose your work because of that so if you want to avoid that you can register for [excalidraw+ plan](https://plus.excalidraw.com/pricing) 

i agree. it'd be helpful to show a notification or toast when storage usage reaches around 80-90% of the total limit. 

so users can be informed before hitting the maximum capacity.

**Comment by bgevko** [comment]
I took a shot at patching this up in [PR 8890](https://github.com/excalidraw/excalidraw/pull/8890)
